Tetris/main.cpp

```diff
@@ -0,0 +1,21 @@
+#include <QGuiApplication>
+#include <QQmlApplicationEngine>
+
+int main(int argc, char *argv[]) {
+  QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+
+  QGuiApplication app(argc, argv);
+
+  QQmlApplicationEngine engine;
+  const QUrl url(QStringLiteral("qrc:/main.qml"));
+  QObject::connect(
+      &engine, &QQmlApplicationEngine::objectCreated, &app,
+      [url](QObject *obj, const QUrl &objUrl) {
+        if (!obj && url == objUrl)
```

Comment:
`&& url == objUrl` - why this additional check is needed?

Reply:
I have fixed all except:
if (!obj && url == objUrl).
It`s auto generation in QT creator. It already was there, when I created a project.